flink-tests/src/test/java/org/apache/flink/test/checkpointing/AutoRescalingITCase.java

```diff
@@ -411,7 +413,7 @@ public void testCheckpointRescalingWithKeyedAndNonPartitionedState() throws Exce
             // clear the CollectionSink set for the restarted job
             CollectionSink.clearElementsSet();
 
-            waitForOneMoreCheckpoint(jobID, cluster.getMiniCluster());
+            waitForNewCheckpoint(jobID, cluster.getMiniCluster());
```

Comment:
I guess the scenario can happen in this test as well because it's almost the same test implementation as in `#testCheckpointRescalingKeyedState` :thinking:

Reply:
I'm wondering whether the redundant code could be removed here. But that's probably a bit out-of-scope for this issue.

Reply:
Do you mean `createJobGraphWithKeyedState` and `createJobGraphWithKeyedAndNonPartitionedOperatorState` have redundant code ? Or `testCheckpointRescalingWithKeyedAndNonPartitionedState` and `testCheckpointRescalingKeyedState`?


I checked them, they have a lot of differences in details. Such as:

- Source is different 
- The parallelism and MaxParallelism is fixed parallelism for `NonPartitionedOperator`

I will check could they extract some common code later. If yes, I can submit a hotfix PR and cc you.

Reply:
I created a [PR](https://github.com/1996fanrui/flink/pull/9) to show what I had in mind for the code redundancy reduction

Reply:
It seems out-of-scope of this JIRA. Would you mind if we refactor it in a hotfix PR?

Your code is ready. After this PR, you can submit a official PR, and I can help review. WDYT?